File: 0001-two-sum/0001-two-sum.go
func twoSum(nums []int, target int) []int {
    freq := make(map[int]int)

    for k,v := range nums {
        if _, ok := freq[target - v]; ok {
            return []int{freq[target - v], k}
        } else {
            freq[v] = k
        }
    }

    return []int{0,0}
}

File: 0001-two-sum/0001-two-sum.rb
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}
def two_sum(nums, target)
    freq = {}
    nums.each_with_index do |num, index|
        if freq[target - num]
            return [freq[target-num],index]
        else
            freq[num] = index
        end
    end
end